ocs_ci/deployment/helpers/hypershift_base.py

```
@@ -1146,38 +1150,71 @@ def download_hosted_cluster_kubeconfig(
-        try:
-            with config.RunWithProviderConfigContextIfAvailable():
-                if from_hcp:
+        def _kubeconfig_ready():
+            return (
+                os.path.isfile(kubeconfig_path) and os.stat(kubeconfig_path).st_size > 0
+            )
+
+        with config.RunWithProviderConfigContextIfAvailable():
+            if from_hcp:
+                try:
                     exec_cmd(
```

Ruff (0.16.2)

[error] Function call with `shell=True` parameter identified, security issue

(S604)